contracts/integrations/test/framework-unit-tests/function_assertion_test.ts

```diff
@@ -0,0 +1,125 @@
+import { blockchainTests, constants, expect, filterLogsToArguments, hexRandom } from '@0x/contracts-test-utils';
+import { BigNumber, generatePseudoRandom256BitNumber } from '@0x/utils';
+import { TransactionReceiptWithDecodedLogs } from 'ethereum-types';
+
+import { artifacts, TestFrameworkContract, TestFrameworkSomeEventEventArgs, TestFrameworkEvents } from '../../src';
+import { FunctionAssertion, Result } from '../utils/function_assertions';
+
+blockchainTests.resets('FunctionAssertion Unit Tests', env => {
+    let exampleContract: TestFrameworkContract;
+
+    before(async () => {
+        exampleContract = await TestFrameworkContract.deployFrom0xArtifactAsync(
+            artifacts.TestFramework,
+            env.provider,
+            env.txDefaults,
+            artifacts,
+        );
+    });
+
+    describe('runAsync', () => {
+        it('should call the before function with the provided arguments', async () => {
+            let sideEffectTarget = constants.ZERO_AMOUNT;
+            const assertion = new FunctionAssertion(exampleContract.noEffect, {
+                before: async (inputValue: BigNumber) => {
+                    sideEffectTarget = inputValue;
+                },
+                after: async (beforeInfo: any, result: Result, returnValue: BigNumber) => {},
+            });
+
+            const randomInput = generatePseudoRandom256BitNumber();
```

Comment:
Optional: We have a `getRandomInteger()` in `@0x/contracts-test-utils` if you'd rather use that to reduce our dependency on `/package` stuff.

Reply:
didn't know that existed 🙊